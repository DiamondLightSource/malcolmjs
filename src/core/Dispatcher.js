/*
 * React.js Starter Kit
 * Copyright (c) 2014 Konstantin Tarkus (@koistya), KriaSoft LLC.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import Flux from 'flux';
import PayloadSources from '../constants/PayloadSources';
import assign from 'react/lib/Object.assign';

/**
 * A singleton that operates as the central hub for application updates.
 * For more information visit https://facebook.github.io/flux/
 */
let Dispatcher = assign(new Flux.Dispatcher(), {

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the server.
   */
  handleServerAction(action) {
    var payload = {
      source: PayloadSources.SERVER_ACTION,
      action: action
    };
    this.dispatch(payload);
  },

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the view.
   */
  handleViewAction(action) {
    var payload = {
      source: PayloadSources.VIEW_ACTION,
      action: action
    };
    this.dispatch(payload);
  }

});

export default Dispatcher;
