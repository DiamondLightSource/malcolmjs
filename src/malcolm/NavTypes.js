const NavTypes = {
  Block: 'block',
  Attribute: 'attribute',
  Info: 'info',
  Palette: 'palette',
  SubElement: 'subelement',
};

export default NavTypes;
