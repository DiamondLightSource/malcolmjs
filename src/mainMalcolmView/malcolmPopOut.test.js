import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import configureStore from 'redux-mock-store';
import MalcolmPopOut from './malcolmPopOut.container';
import NavTypes from '../malcolm/NavTypes';

describe('MalcolmPopOut', () => {
  let shallow;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
  });

  it('renders correctly for block', () => {
    const state = {
      malcolm: {
        parentBlock: 'test1',
        blocks: {
          test1: {
            name: 'Test main view',
          },
        },
        navigation: {
          navigationLists: [{ path: 'test1', navType: NavTypes.Block }],
        },
      },
    };

    const mockStore = configureStore();

    const wrapper = shallow(<MalcolmPopOut store={mockStore(state)} />);
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('renders correctly for info', () => {
    const state = {
      malcolm: {
        parentBlock: 'test1',
        childBlock: '.info',
        mainAttribute: 'attr1',
        blocks: {
          test1: {
            name: 'Test main view',
          },
        },
        navigation: {
          navigationLists: [
            { path: 'test1', navType: NavTypes.Block },
            { path: 'attr1', navType: NavTypes.Attribute },
            { path: '.info', navType: NavTypes.Info },
          ],
        },
      },
    };

    const mockStore = configureStore();

    const wrapper = shallow(<MalcolmPopOut store={mockStore(state)} />);
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
