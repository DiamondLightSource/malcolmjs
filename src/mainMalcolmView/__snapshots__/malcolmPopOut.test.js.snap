// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MalcolmPopOut renders correctly 1`] = `
<div
  className="Connect-MalcolmPopOut--container-1"
>
  <WithStyles(AppBar)
    position="static"
  >
    <WithStyles(Toolbar)
      className="Connect-MalcolmPopOut--toolbar-2"
    >
      <WithStyles(Typography)
        className="Connect-MalcolmPopOut--title-3"
        color="inherit"
        variant="title"
      >
        Test main view
      </WithStyles(Typography)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
  <Connect(WithStyles(BlockDetails))
    parent={true}
  />
</div>
`;
